src/reducers/signupReducer.js

```diff
@@ -0,0 +1,27 @@
+import { signupType } from '../actions/signupActions';
+import initialState from '../store/initialState';
+
+
+/**
+ * @description - signup reducer function
+ * @param {object} [state=initialState] - Initial state to load
+ * @param {Function} action - Action to dispatch
+ * @returns {object}
+ */
+export default (state = initialState, action ) => {
+ 
+  switch (action.type) {
+    case signupType.loading:
+      return { loading: action.data };
+    case signupType.error:
+      return { errors: action.data.errors, loading: action.data.loading };
+    case signupType.clear_errors:
```

Comment:
#### Nitpick
- Why the introduction of so many error action types?

Reply:
@davdwhyte87 can you please explain this error action types? I think they are too many

Reply:
They handle different cases. I may want to stop loading value at some point, and at other points, I just want to change just error values

Reply:
I have made changes. It's good we have a pattern. Here are the changes.
https://github.com/andela/apollo-ah-frontend/blob/feature%2F164198142%2Fsignup/src%2Freducers%2FsignupReducer.js

https://github.com/andela/apollo-ah-frontend/blob/feature%2F164198142%2Fsignup/src%2Factions%2FtypeGenerator.js

https://github.com/andela/apollo-ah-frontend/blob/feature%2F164198142%2Fsignup/src%2Factions%2FsignupActions.js

And the tests for the actions and reducers now point to the 3 defined types.